### PR TITLE
add avro-python3

### PR DESCRIPTION
**Pull request by ocefpaf**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@conda-forge/staged-recipes team, this is noarch and there is a missing dep on Windows. It should be OK to keep it noarch. Please review!

**Comment by dopplershift** [comment]
Did you see:
> # Deprecation Notice
[avro-python3](https://pypi.org/project/avro-python3/) is no longer maintained and is deprecated in favor of [avro](https://pypi.org/project/avro/). Avro is maintained and has full support for recent versions of Python.

**Comment by ocefpaf** [comment]
Sure. Deprecated, but I guess some packages still have it as a dependency in the metadata and pip check will fail without this one until there is a new release that specifies just avro.

File: recipes/avro-python3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python >=3.6
+    - avro
+    - isort
+    - pip
+    - pycodestyle
```
Comment (dopplershift) [inline recipes/avro-python3/meta.yaml:24]: Are `isort` and `pycodestyle` really needed to build the package?

Comment (ocefpaf) [reply recipes/avro-python3/meta.yaml:24]: Sadly yes. Some people don't know how to package dev deps properly :-(

**Review by dopplershift — APPROVED** [review]

**Pull request merged**